src/components/Disclosure/Disclosure.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+
+import {block} from '../utils/cn';
+import {isOfType} from '../utils/isOfType';
+
+import {DisclosureProvider} from './DisclosureContext';
+import {DisclosureDetails} from './DisclosureDetails/DisclosureDetails';
+import {DefaultDisclosureSummary, DisclosureSummary} from './DisclosureSummary/DisclosureSummary';
+
+import './Disclosure.scss';
+
+export const b = block('disclosure');
+
+export type DisclosureSize = 'm' | 'l' | 'xl';
+export type DisclosureArrowPosition = 'left' | 'right';
+
+export interface DisclosureComposition {
+    Summary: typeof DisclosureSummary;
+}
+
+export interface DisclosureProps {
+    /** Disclosure size */
+    size: DisclosureSize;
+    /** Disabled state */
+    disabled?: boolean;
+    /** Default opening state */
+    defaultExpanded?: boolean;
```

Comment:
Component has only uncontrolled mode?

Reply:
For now, yes. It takes initial condition and after that handles state on its own.

Reply:
It is a widely used functional I suppose. Could u add this props?